tests/test_tempfile.py

```diff
@@ -0,0 +1,225 @@
+            stf._fp._rolled = False
+            stf._fp._max_size = 20
+            rollover_called = False
+            original_rollover = stf._fp.rollover
+
+            def fake_rollover() -> None:
+                nonlocal rollover_called
+                rollover_called = True
+                return original_rollover()
+
+            stf._fp.rollover = fake_rollover
+            await stf.writelines([b"hello", b"world"])
+            await stf.seek(0)
+
+            assert await stf.read() == b"helloworld"
+
+            await stf.writelines([b"1234567890123456"])
+            assert rollover_called
+
+    async def test_closed_state(self) -> None:
+        async with SpooledTemporaryFile[bytes](max_size=10) as stf:
+            assert not stf.closed
+
+        assert stf.closed
+
+
+class TestTemporaryDirectory:
+    async def test_context_manager(self) -> None:
+        async with TemporaryDirectory() as td:
+            td_path = pathlib.Path(td)
+            assert td_path.exists() and td_path.is_dir()
+
+            file_path = td_path / "test.txt"
+            file_path.write_text("temp dir test", encoding="utf-8")
+            assert file_path.exists()
+
+        assert not td_path.exists()
+
+    async def test_cleanup_method(self) -> None:
+        td = TemporaryDirectory()
+        td_str = await td.__aenter__()
+        td_path = pathlib.Path(td_str)
+
+        file_path = td_path / "file.txt"
+        file_path.write_text("cleanup test", encoding="utf-8")
+
+        await td.cleanup()
+        assert not td_path.exists()
+
+    async def test_exception_handling(self) -> None:
+        async with TemporaryDirectory() as td:
+            td_path = pathlib.Path(td)
+            assert td_path.exists() and td_path.is_dir()
+
+        assert not td_path.exists()
+
+        with pytest.raises(FileNotFoundError):
+            (td_path / "nonexistent.txt").write_text("should fail", encoding="utf-8")
+
+
+@pytest.mark.parametrize(
+    "suffix, prefix, text, content",
+    [
+        (".txt", "mkstemp_", True, "mkstemp"),
+        (b".txt", b"mkstemp_", False, b"mkstemp"),
+    ],
+)
+async def test_mkstemp(
+    suffix: AnyStr,
+    prefix: AnyStr,
+    text: bool,
+    content: AnyStr,
+) -> None:
+    fd, path = await mkstemp(suffix=suffix, prefix=prefix, text=text)
+
+    assert isinstance(fd, int)
+    if text:
+        assert isinstance(path, str)
+    else:
+        assert isinstance(path, bytes)
+
+    if text:
+        with os.fdopen(fd, "w", encoding="utf-8") as f:
+            f.write(content)
+        with open(path, encoding="utf-8") as f:
+            read_content = f.read()
+    else:
+        with os.fdopen(fd, "wb") as f:
+            f.write(content)
+        with open(os.fsdecode(path), "rb") as f:
+            read_content = f.read()
+
+    assert read_content == content
+
+    os.remove(path)
+
+
+@pytest.mark.parametrize("prefix", [b"mkdtemp_", "mkdtemp_"])
+async def test_mkdtemp(prefix: AnyStr) -> None:
```

Comment:
The typevar is unwarranted here.
```suggestion
async def test_mkdtemp(prefix: str | bytes) -> None:
```